adagrams/game.py

```diff
@@ -1,11 +1,99 @@
+import random
+
 def draw_letters():
-    pass
+
+    letter_distribution = {
+    'A': 9, 
+    'B': 2, 
+    'C': 2, 
+    'D': 4, 
+    'E': 12, 
+    'F': 2, 
+    'G': 3, 
+    'H': 2, 
+    'I': 9, 
+    'J': 1, 
+    'K': 1, 
+    'L': 4, 
+    'M': 2, 
+    'N': 6, 
+    'O': 8, 
+    'P': 2, 
+    'Q': 1, 
+    'R': 6, 
+    'S': 4, 
+    'T': 6, 
+    'U': 4, 
+    'V': 2, 
+    'W': 2, 
+    'X': 1, 
+    'Y': 2, 
+    'Z': 1
+}
+    ten_rand_letters = []
+
+    while len(ten_rand_letters) < 10:
+        random_letter =random.choice(list(letter_distribution.keys()))   
+        
+        if ten_rand_letters.count(random_letter) < letter_distribution[random_letter]:
+            ten_rand_letters.append(random_letter)
+    return ten_rand_letters
 
 def uses_available_letters(word, letter_bank):
-    pass
+
+    upper_word = word.upper()
+    
+    for letter in upper_word:
+        if letter not in letter_bank or letter_bank.count(letter) < upper_word.count(letter):
+            return False
+        else:
+            continue
+    return True
 
 def score_word(word):
-    pass
+
+    score = 0
+    letter_values = {
+            ("A", "E", "I", "O", "U", "L", "N", "R", "S", "T") : 1, 
+            ("D", "G") : 2,
+            ("B", "C", "M", "P") : 3,
+            ("F", "H", "V", "W", "Y") :4,
+            ("K") : 5,
+            ("J", "X") : 8,
+            ("Q", "Z") : 10
+            }
+    
+    upper_word = word.upper()
+    for letter in upper_word:
+        for key in letter_values.keys():
+            if letter in key:
```

Comment:
By organizing the score table to have a tuple of tiles as the key, we are effectively forced to iterate over the individual values in the score table keys to find our tile. This is not using the benefits of a dictionary to its utmost. When storing data in a dictionary, we should put the data that we expect to frequently lookup by as the key rather than in the value.

If our table stored the letter tile as the key and the score for that tile as the value, we would be able to iterate over the letters in the word, and look up the score for each letter in constant time.

This results in more keys appearing in the table, but for an arbitrary score system, this could still happen with the current structure. If each tile had a distinct score, then we would have as many keys as tiles.

There are always competing concerns between using space efficiently, and runtime performance. But one thing we can definitely say about using dicts is that if we've chosen to use a dict, we should structure the data so that we can look up keys directly.